src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -406,7 +407,7 @@ def merge_type(a: DataType, b: DataType, name: Optional[str] = None) -> DataType
         return a
 
 
-def python_type_str_to_object(tp_str: str) -> Type:
+def python_type_str_to_object(tp_str: str, object_type: str) -> Type:
```

Comment:
Should we assert on the expected `object_type`?

Reply:
Basically checking if `object_type` is of the expected types (I imagine only UDXF and sp). Are there other use cases that might call this?

Reply:
We also call it with `python_type_to_snow_type` in `src/snowflake/snowpark/dataframe_na_functions.py` which is none of the objects in `TempObjectType` enums. Right now, I'm calling this function with `object_type=""`

Reply:
Another option is to just pass in `is_return_type_of_sproc` and default to `False`, and if we need more than `sproc`, we could refactor it to take in `object_type` 🤔 . Will this make it clearer?

Reply:
sounds like a better alternative. will work on it.